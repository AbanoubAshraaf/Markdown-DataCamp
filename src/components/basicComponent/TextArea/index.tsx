import React from 'react';
import { TextInput, StyleSheet } from 'react-native';

interface TextAreaProps {
  value: string;
  onChangeText: (text: string) => void;
  placeholder: string;
  style?: any;
}

const TextArea: React.FC<TextAreaProps> = ({ value, onChangeText, placeholder, style }) => {
  return (
    <TextInput
      style={style}
      multiline
      placeholder={placeholder}
      value={value}
      onChangeText={onChangeText}
    />
  );
};

export default TextArea;
