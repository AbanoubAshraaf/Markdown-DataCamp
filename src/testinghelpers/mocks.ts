export const mockedMarkDownText = 'markdown text'

export const mockedPortraitDimensions = {
    width: 400,
    height: 800,
    scale: 0,
    fontScale: 0,
};

export const mockedLandScapeDimensions = {
    width: 800,
    height: 400,
    scale: 0,
    fontScale: 0,
};