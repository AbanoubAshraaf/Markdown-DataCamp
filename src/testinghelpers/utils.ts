import { RenderResult } from "@testing-library/react-native";

interface Screen extends RenderResult {
    isDetached?: boolean;
}

export const createSnapShot = (screen: Screen) => {
    expect(screen).toMatchSnapshot();
  };