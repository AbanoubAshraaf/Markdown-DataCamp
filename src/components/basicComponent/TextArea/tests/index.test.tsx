import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import TextArea from '..';
import { createSnapShot } from 'src/testinghelpers/utils';

describe('TextArea', () => {
  it('renders correctly and calls onChangeText when text is changed', () => {
    const onChangeTextMock = jest.fn();
    const { getByPlaceholderText } = render(
      <TextArea
        value=""
        onChangeText={onChangeTextMock}
        placeholder="Enter your markdown here..."
      />
    );

    const input = getByPlaceholderText('Enter your markdown here...');
    fireEvent.changeText(input, 'New text');

    expect(onChangeTextMock).toHaveBeenCalledWith('New text');
    createSnapShot(screen)
  });
});
