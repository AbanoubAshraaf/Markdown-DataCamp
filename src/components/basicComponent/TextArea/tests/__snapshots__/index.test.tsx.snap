// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly and calls onChangeText when text is changed 1`] = `
<TextInput
  multiline={true}
  onChangeText={
    [MockFunction] {
      "calls": [
        [
          "New text",
        ],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="Enter your markdown here..."
  value=""
/>
`;
